frontend/src/components/pages/agents/details/chat/chat-code-block.tsx

```diff
@@ -0,0 +1,121 @@
+import { Box } from '@chakra-ui/react';
+import { CopyButton } from '@redpanda-data/ui';
+import { Prism as SyntaxHighlighter, type SyntaxHighlighterProps } from 'react-syntax-highlighter';
+import prism from 'react-syntax-highlighter/dist/cjs/styles/prism/prism';
+import { nightOwlTheme } from './night-owl-theme';
+
+export interface CodeBlockProps extends Omit<SyntaxHighlighterProps, 'children'> {
+  language: string;
+  codeString: string;
+  highlightLines?: [number, number];
+  removeLineBreaksFromCopy?: boolean;
+  showLineNumbers?: boolean;
+  showCopyButton?: boolean;
+  /**
+   * @deprecated this setting is no longer relevant, the scroll will appear if required
+   */
+  showScroll?: boolean;
+  theme?: 'dark' | 'light';
+}
+
+export const CodeBlock = ({
+  language,
+  codeString,
+  highlightLines,
+  removeLineBreaksFromCopy = false,
+  showLineNumbers = false,
+  showCopyButton = true,
+  theme = 'dark',
+  ...otherProps
+}: CodeBlockProps) => {
+  const shouldHighlightLine = (index: number) => {
+    if (!highlightLines) {
+      return false;
+    }
+    const lineNumber = index + 1;
+    const inRange = lineNumber >= highlightLines[0] && lineNumber <= highlightLines[1];
+    return inRange;
+  };
+
+  const backgroundColor = theme === 'dark' ? '#011627' : '#F5F2F0';
```

Comment:
The same here, I'd prefer if we use some better defined colors from named constants or ideally from the palette.

Reply:
I had to leave it like this because it's a rather specific color that does not belong to the palette and it's only for the code block, I moved it to the night owl theme that we are setting for dark mode, at least they are separate constants